src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PremisesService.kt

```diff
@@ -91,4 +95,74 @@ class PremisesService(
 
       return success(lostBedsEntity)
     }
+
+  fun createNewPremises(requestBody: NewPremises) = validated<PremisesEntity> {
+
+    // TODO: dummy data for now - remove asap
+    val apAreaEntity = ApAreaEntity(
+      id = UUID.randomUUID(),
+      name = "arbitrary_ap_area",
+      identifier = "arbitrary_identifier",
+      probationRegions = mutableListOf()
+    )
+
+    val probationRegion = ProbationRegionEntity(
+      id = UUID.fromString("afee0696-8df3-4d9f-9d0c-268f17772e2c"), // Wales in db
+      name = "arbitrary_probation_region",
+      apArea = apAreaEntity,
+      premises = mutableListOf()
+    )
+    // end of dummy data
+
+    if (requestBody.address.isEmpty()) {
+      return generalError("Please provide an address")
+    }
+
+    if (requestBody.postcode.isEmpty()) {
+      return generalError("Please provide a postcode")
+    }
+
+    if (requestBody.service.isEmpty()) {
+      return generalError("Please provide a service type")
+    }
+
+    try {
+      UUID.fromString(requestBody.localAuthorityId)
+    } catch (exception: Exception) {
+      return generalError("Please provide a local authority ID in valid UUID format")
+    }
+
+    // TODO: on the next iteration the identifier and name will be retrieved by the id
+    val localAuthorityArea = LocalAuthorityAreaEntity(
+      id = UUID.fromString(requestBody.localAuthorityId),
+      identifier = "arbitrary_identifier",
+      name = "arbitrary_local_authority_area",
+      premises = mutableListOf()
+    )
+
+    val premisesName = when (requestBody.name.isNullOrEmpty()) {
+      true -> "Unknown"
+      false -> requestBody.name
+    }
+
+    val premisesEntity = premisesRepository.save(
+      PremisesEntity(
+        id = UUID.randomUUID(),
+        name = premisesName,
+        apCode = "CAS3_default_apcode",
```

Comment:
I would think we'd want to change this to be nullable in the DB and kotlin models rather than providing it a sentinel value?